internal/pkg/otel/configtranslate/otelconfig.go

```diff
@@ -0,0 +1,366 @@
+	receiverName := fmt.Sprintf("%s%s", OtelNamePrefix, unitID)
+	return otelcomponent.NewIDWithName(receiverType, receiverName)
+}
+
+// getExporterID returns the exporter id for the given exporter type and output name.
+func getExporterID(exporterType otelcomponent.Type, outputName string) otelcomponent.ID {
+	exporterName := fmt.Sprintf("%s%s", OtelNamePrefix, outputName)
+	return otelcomponent.NewIDWithName(exporterType, exporterName)
+}
+
+// getCollectorConfigForComponent returns the Otel collector config required to run the given component.
+// This function returns a full, valid configuration that can then be merged with configurations for other components.
+func getCollectorConfigForComponent(comp *component.Component, info info.Agent) (*confmap.Conf, error) {
+	outputQueueConfig := getOutputQueueConfig(comp)
+	receiversConfig, err := getReceiversConfigForComponent(comp, info, outputQueueConfig)
+	if err != nil {
+		return nil, err
+	}
+	exportersConfig, err := getExportersConfigForComponent(comp)
+	if err != nil {
+		return nil, err
+	}
+	pipelineID, err := getPipelineID(comp)
+	if err != nil {
+		return nil, err
+	}
+	pipelinesConfig := map[string]any{
+		pipelineID.String(): map[string][]string{
+			"exporters": maps.Keys(exportersConfig),
+			"receivers": maps.Keys(receiversConfig),
+		},
+	}
+
+	fullConfig := map[string]any{
+		"receivers": receiversConfig,
+		"exporters": exportersConfig,
+		"service": map[string]any{
+			"pipelines": pipelinesConfig,
+		},
+	}
+	return confmap.NewFromStringMap(fullConfig), nil
+}
+
+// getReceiversConfigForComponent returns the receivers configuration for a component. Usually this will be a single
+// receiver, but in principle it could be more.
+func getReceiversConfigForComponent(comp *component.Component, info info.Agent, outputQueueConfig map[string]any) (map[string]any, error) {
+	receiverType, err := getReceiverTypeForComponent(comp)
+	if err != nil {
+		return nil, err
+	}
+	// this is necessary to convert policy config format to beat config format
+	defaultDataStreamType, err := getDefaultDatastreamTypeForComponent(comp)
+	if err != nil {
+		return nil, err
+	}
+
+	// get inputs for all the units
+	// we run a single receiver for each component to mirror what beats processes do
+	var inputs []map[string]any
+	for _, unit := range comp.Units {
+		if unit.Type == client.UnitTypeInput {
+			unitInputs, err := getInputsForUnit(unit, info, defaultDataStreamType)
+			if err != nil {
+				return nil, err
+			}
+			inputs = append(inputs, unitInputs...)
+		}
+	}
+
+	receiverId := getReceiverID(receiverType, comp.ID)
+	// Beat config inside a beat receiver is nested under an additional key. Not sure if this simple translation is
+	// always safe. We should either ensure this is always the case, or have an explicit mapping.
+	beatName := strings.TrimSuffix(receiverType.String(), "receiver")
+	beatDataPath := filepath.Join(paths.Home(), "run", comp.ID)
```

Comment:
This could be simplified using https://github.com/elastic/elastic-agent/blob/38bcd03d6bfe043d1e9cbbe393092dd9600fd14e/internal/pkg/agent/application/paths/common.go#L199-L201
```suggestion
	beatDataPath := filepath.Join(paths.Run(), comp.ID)
```